common/src/main/java/com/ably/tracking/common/workerqueue/Worker.kt

```diff
@@ -1,12 +1,14 @@
-package com.ably.tracking.subscriber.workerqueue
+package com.ably.tracking.common.workerqueue
 
 import com.ably.tracking.common.ResultCallbackFunction
-import com.ably.tracking.subscriber.Properties
 
 /**
  * A [Worker] interface represents workers which execute work inside [WorkerQueue].
+ * Params:
+ * P - the type of properties used by this worker as both input and output
+ * S - the type of specification used to post another worker back to the queue
  */
-internal interface Worker {
+interface Worker<P : Properties, S> {
```

Comment:
WDYT about using full words for the type parameters? I read an article that suggested using full names because one-letter abbreviations introduce unnecessary cognitive load while you're analyzing code 😉 Instead of using `P` and `S` and having this docs annotation about what they represent, we could use `PropertiesType` and `WorkerSpecificationType` which should make it self explanatory both here and below in places where you use it 🤔 
```suggestion
interface Worker<PropertiesType : Properties, WorkerSpecificationType> {
```

Reply:
This doesn't feel natural, but I see no reason not to do it 😮 
01a23c27660695394f7e30f8660fec637d86992f

Reply:
Actually, I didn't mean to rename the `Properties` to something different in order to change this type declaration 😅 I wanted to change the `P` to `PropertiesType` to make it explicit that this is a "generic type" and it extends "Properties" 😉 So basically to take the original name and add the `Type` suffix.

Reply:
ed24ab40047589268666b59f404bbb32077c81f2

Reply:
I think you've missed [one place](https://github.com/ably/ably-asset-tracking-android/pull/786/files#diff-295145610258ad576bb05bd11957d365028c45d65d43407467d6e9dfa88503e4R42-R43) 😉

Reply:
61f05aa451463edbd94461b45ce710de03beb0ed